### PR TITLE
Feature/69_unit_tests_assets

### LINKED ISSUE #69: Refactor ManagementClientTests Architecture
### Motivation

* Currently, the tests are hard to maintain, read, and basically dependent on the live project in Kontent if new features are added and bugs are fixed.
* Re-generation requires there steps - Re-run ManagementClientTests with `private static readonly TestUtils.TestRunType _runType = TestUtils.TestRunType.MockFromFileSystem;`, after tests are green, commit changes in `~/Kentico.Kontent.Management.Tests/Data/*`

Come up with a better approach that would test the SDK functionality without necessity to pre-record requests and reposnse.

### Proposed solution

These are just possible enhancements:
* [x] Do now generate hash under `Data` folder - identify mock responses just by test name - so that update does not generate folder removal and addiction - and this pair is hard to distinguish
* [ ] Automate regeneration of the mock files
* [ ] Hardcode the JSONs and avoid auto-generation completely

### Additional context

* We might use [MAPI Swagger definition](https://kcdmaster.blob.core.windows.net/api-reference-specifications/management_api_v2.json) and use it for mock server -  (if URL does not work - reach out to @Simply007)

### Related issues

#79 #66 #64

### PR DESCRIPTION
**Pull request by gormal**
### Motivation

Fixes #69

Additionally, Update* methods have been removed for assets as they don't make much sense.

### CODE REVIEW

File: Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/AssetTests.cs [hunk 1]
```diff
@@ -138,216 +189,453 @@ public async Task CreateAssetAsync_CreatesDynamicAssetModel()
...
-        
+            var client = _fileSystemFixture.CreateMockClientWithResponse("File.json", "Asset.json");
+
             var expected = GetExpectedStronglyTypedAssetModel();
-        
-            var createModel = new AssetCreateModel<ComplexTestModel>
+
+            MemoryStream stream = new MemoryStream(Encoding.UTF8.GetBytes("Hello world from CM API .NET SDK"));
+            string fileName = "Hello.txt";
+            string contentType = "text/plain";
```
Comment (Sevitas) [inline Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/AssetTests.cs:209]: few other places aswell

```suggestion
            var stream = new MemoryStream(Encoding.UTF8.GetBytes("Hello world from CM API .NET SDK"));
            var fileName = "Hello.txt";
            var contentType = "text/plain";
```

**Review by Sevitas — CHANGES_REQUESTED** [review]
Just a few nitpicks, resolve/merge it's up to you

File: Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/AssetTests.cs (hunk 1 above)
Comment (gormal) [reply Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/AssetTests.cs]: Fixed

File: Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/AssetTests.cs [hunk 2]
```diff
@@ -10,6 +10,8 @@
 using Kentico.Kontent.Management.Extensions;
 using Kentico.Kontent.Management.Models.Assets;
 using Kentico.Kontent.Management.Models.StronglyTyped;
+using System.IO;
+using System.Text;
```
[the remark this answers is not in the document]
Comment (gormal) [reply Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/AssetTests.cs:14]: Fixed

File: Kentico.Kontent.Management/Models/Assets/AssetUpsertModel.cs [hunk 3]
```diff
@@ -11,17 +11,11 @@ namespace Kentico.Kontent.Management.Models.Assets
     /// </summary>
```
[the remark this answers is not in the document]
Comment (gormal) [reply Kentico.Kontent.Management/Models/Assets/AssetUpsertModel.cs:9]: Fixed

**Comment by gormal** [comment]
Thanks for the review, minor issues were fixed.

**Pull request merged**